### ISSUE TITLE
What profiles does this now actually support?

### ISSUE THREAD

**Issue by erazortt** [erazortt/DoViBaker#12]
I think you guys (@Asd-g @quietvoid) have made some tests with sources other than profle 7, right?

**Comment by quietvoid** [comment]
Yes, only profile 5 still has desaturated colours compared to expected image.
I noticed the RGB to LMS coeffs aren't used, too.

Otherwise 8.4 is fine.

**Comment by erazortt** [comment]
In which curcumstances am I supposed to use the LSM coeffs? What processes LSM output?

**Comment by quietvoid** [comment]
If I refer to `libplacebo`:

```c
// Colorspace transformation metadata
float nonlinear_offset[3];      // input offset ("ycc_to_rgb_offset")
struct pl_matrix3x3 nonlinear;  // before PQ, also called "ycc_to_rgb"
struct pl_matrix3x3 linear;     // after PQ, also called "rgb_to_lms"
```

I can't say if this being missing is the reason for the incorrect colours. It seems to affect chroma only.
Though as it's RGB, I guess it is because of a missing step.

It does PQ RGB > linearization > RGB to LMS matrix > delinearization (back to PQ).
So this seems to go right after the conversion to RGB.

**Comment by ValZapod** [comment]
>Yes, only profile 5 still has desaturated colours compared to expected image.

Ha? DoviBaker is not for profile 5's IPTPQc2.

**Comment by ValeZAA** [comment]
>Yes, only profile 5 still has desaturated colours compared to expected image.

I do not get it, can profile 5 IPTPQc2 have actually FEL?? But even if it can, it does not matter, there is no content like that in the wild.

**Comment by graham8** [comment]
> > Yes, only profile 5 still has desaturated colours compared to expected image.
> 
> I do not get it, can profile 5 IPTPQc2 have actually FEL?? But even if it can, it does not matter, there is no content like that in the wild.

Profile 5 does not have FEL.  It's single-layer.

@erazortt Can't this be closed, now?

**Comment by ValeZAA** [comment]
Erm, no, Profile 5 is 2 layers, it is just that 2nd layer is integrated as RPU into first layer. The queston is whether we can declare FEL in RPU. I doubt it.

**Comment by graham8** [comment]
Sorry.  You are wrong.  See the DV Profiles and Levels doc.  FEL means there is video information in a second layer.

**Comment by ValeZAA** [comment]
>EL means there is video information in a second layer.

Not always. 2nd layer may have no video information in the case of MEL. Or just have fake padding, zeroed out hevc.

>the DV Profiles and Levels doc.

You mean dual layer specification from ETSI.

**Comment by graham8** [comment]
I said **FEL** means there is video infomation.  MEL has no video information - agreed.

https://dolby.my.salesforce.com/sfc/p/#700000009YuG/a/4u000000l6G4/4R18riPaaW3gxpVx7XwyQLdEITLFjB.w.Si0LoQR5j8

**Comment by ValeZAA** [comment]
> I said **FEL** means there is video infomation. MEL has no video information - agreed.
> 
> https://dolby.my.salesforce.com/sfc/p/#700000009YuG/a/4u000000l6G4/4R18riPaaW3gxpVx7XwyQLdEITLFjB.w.Si0LoQR5j8

No, FEL can be NLQ only https://github.com/haasn/libplacebo/pull/286

Technically yes, NLQ is just first stage, you need video data ideally.
We need Haasn to answer what he meant.

**Comment by graham8** [comment]
So what you (or @hassn) are saying is that P5 can have (trivial) NLQ coefficients?  I've never come across that.  And surely it can safely be ignored.  Trivial NLQs indicate MEL.  Non-trivial NLQs would have nothing to do if there's no EL video.

**Comment by ValeZAA** [comment]
> > I said **FEL** means there is video infomation. MEL has no video information - agreed.
> > https://dolby.my.salesforce.com/sfc/p/#700000009YuG/a/4u000000l6G4/4R18riPaaW3gxpVx7XwyQLdEITLFjB.w.Si0LoQR5j8
> 
> No, FEL can be NLQ only [haasn/libplacebo#286](https://github.com/haasn/libplacebo/pull/286)
> 
> Technically yes, NLQ is just first stage, you need video data ideally. We need Haasn to answer what he meant.

This is merged now; also mpv now supports source-dynamic which allows dynamic tonemapping using scene changing static HDR10 metadata, this is how Apple TV console does it.
